Tribler/community/multichain/community.py

```diff
@@ -282,6 +283,75 @@ def get_statistics(self, public_key=None):
         return statistics
 
+    @blocking_call_on_reactor_thread
+    def get_nodes(self, public_key=None, neighbor_radius=1):
+        """
+        Returns a dictionary with the neighboring nodes of a certain focus node within a certain radius,
+            regarding the local multichain database
+
+        :param public_key: the public key of the focus node
+        :param neighbor_radius: the radius within which the neighbors have to be returned
+        :return: a dictionary with nodes
+        """
+        if public_key is None:
+            public_key = self.my_member.public_key
+        # TODO: make call instead of using dummy data
+        #list_of_nodes = self.persistence.get_list_of_nodes(public_key, neighbor_radius)
+        list_of_nodes = ["abc", "def", "ghi"]
+        if public_key not in list_of_nodes:
+            list_of_nodes.append(public_key)
+        number_of_nodes = len(list_of_nodes)
+        nodes = []
+        for current in range(number_of_nodes):
+            current_key = list_of_nodes[current]
+            current_statistics = self.get_statistics(current_key)
+            nodes[current] = dict()
+            #nodes[current]["public_key"] = current_key.encode("hex")
+            #nodes[current]["total_up"] = current_statistics["total_up"]
+            #nodes[current]["total_down"] = current_statistics["total down"]
+            nodes[current]["public_key"] = current_key
+            nodes[current]["total_up"] = 0
+            nodes[current]["total_down"] = 0
+        return nodes
+
+    @blocking_call_on_reactor_thread
+    def get_edges(self, nodes=None):
+        """
+        Returns a dictionary with all edges between certain nodes around a certain focus node,
+            regarding the local multichain database
+
+        :param public_key: the public key of the focus node
+        :param nodes: the dictionary of nodes between which the edges have to be returned
+        :return: a dictionary with edges
+        """
+        list_of_nodes = [node["public key"] for node in nodes]
+        # TODO: make call instead of using dummy data
+        # for pair in itertools.combinations(list_of_nodes, 2):
+        #    list_of_edges.append(self.persistence.get_edge(*pair))
+        list_of_edges = [["abc", "def", 0, 0], ["def", "abc", 0, 0], ["abc", "ghi", 0, 0], ["ghi", "abc", 0, 0],
+                         ["abc", "xyz", 0, 0], ["xyz", "abc", 0, 0], ["def", "ghi", 0, 0], ["ghi", "def", 0, 0],
+                         ["def", "xyz", 0, 0], ["xyz", "def", 0, 0], ["ghi", "xyz", 0, 0], ["xyz", "ghi", 0, 0]]
+        number_of_edges = len(list_of_edges)
+        edges = []
+        for current in range(number_of_edges):
+            edges[current] = dict()
+            edges[current]["from"] = list_of_edges[current][0]
+            edges[current]["to"] = list_of_edges[current][1]
+            edges[current]["size"] = self.compute_edge_size(list_of_edges[current][2], list_of_edges[current][3])
```

Comment:
This is more of a discussion than a comment on the code itself. I now understand that size is a ambiguous name; I meant the amount of data in total transported from the "to" to the "from" node. I thought we agreed on calculating the actual graphical size of the edges used in the GUI after the HTTP Response has been received in the GUI end.